### DISCUSSION TITLE
Model pre and post-process in C++ inference ?

### DISCUSSION THREAD

**Discussion by huanidz** [PaddlePaddle/PaddleOCR#7793] [Q&A]
I have converted en_infer recognition model into onnx then tensorrt.
The inference step runs OK but how do i post-process the output layer (softmax) into actual result ?, given the **en_dict**.

Total character in en_dict (from /ppocr/utils/dict/) is 63 while the softmax output have the shape of _N x 12 x 97._
How do i post-process this, is there any post process code in python i can reference to ?

And also, any pre process that affect the size ?

The model that i downloaded:
**Model link**: https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/configs/rec/PP-OCRv3/en_PP-OCRv3_rec.yml

Thanks Paddle team.

**Comment by huanidz** [comment]
i figured it out

**Reply by senstar-hsoleimani** [reply]
@huanidz  I converted en_infer detection, classification, and recognition models to onnx (as it is explained here:https://github.com/PaddlePaddle/PaddleOCR/blob/release/2.6/deploy/paddle2onnx/readme.md). However, I could not load these converted models by openCV (readNetFromOnnx()).
Can you please help me on this? 
I would appreciate that if you share the .onnx model you discussed above.

**Reply by huanidz** [reply]
@senstar-hsoleimani
I'm not sure what error you've encountered, it will be more clear if you show me the log.
For the problem, actually i was using ONNX only once as the middle exchange between Paddle and TensorRT. 
But for loading and inferencing with OpenCV, please make sure you're using right OpenCV version that support that Onnx's opset version.

I don't remember where i left my ONNX model, lol. But if i remember correctly, i just ran a simple command from paddle2onnx so our model should be the same 😄

**Reply by senstar-hsoleimani** [reply]
> @senstar-hsoleimani I'm not sure what error you've encountered, it will be more clear if you show me the log. For the problem, actually i was using ONNX only once as the middle exchange between Paddle and TensorRT. But for loading and inferencing with OpenCV, please make sure you're using right OpenCV version that support that Onnx's opset version.
> 
> I don't remember where i left my ONNX model, lol. But if i remember correctly, i just ran a simple command from paddle2onnx so our model should be the same 😄

Thanks for the reply. I am using opencv 4.5.4
It throws out a memory exception when it load the model (cv::dnn::Net net= cv::dnn::readNet("det.onnx") ):
Unhandled exception at 0x00007FF8AD4F4FD9 in CoreVisionAlg.exe: Microsoft C++ exception: cv::Exception at memory location 0x000000C65BB4EDC0. occurred
